internal/host/host.go

```diff
@@ -1441,14 +1447,29 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 }
 
 func (m *Manager) UnRegisterHost(ctx context.Context, h *models.Host) error {
-	if err := common.DeleteHostFromDB(m.db, h.ID.String(), h.InfraEnvID.String()); err != nil {
-		return err
+	if h.ClusterID == nil {
+		// try to get ClusterId if not provided. (API)
+		err := m.db.Table("hosts").Select("cluster_id").Where("id = ?", h.ID.String()).Scan(&h.ClusterID).Error
+		if err != nil {
+			m.log.Infof("Host %v is not Bound to Cluster.", h.ID)
+		}
 	}
 	if h.ClusterID != nil {
-		if err := m.db.Model(&common.Cluster{}).Where("id = ?", h.ClusterID).Update("trigger_monitor_timestamp", time.Now()).Error; err != nil {
+		var cluster *common.Cluster
+		if err := m.db.Select("status").Take(&cluster, "id = ?", h.ClusterID.String()).Error; err != nil {
+			return fmt.Errorf("failed to get cluster from DB: %+v", err)
+		}
+		if err := common.CanUnbindhost(cluster); err != nil {
```

Comment:
This function `UnRegisterHost` is for deleting hosts, not unbinding them.

If you want this to be run during unbind you should call it from `UnbindHost` if it's checking if a host can be deleted it should have a different name.